test/e2e/jmx/jmx_test.go

```diff
@@ -0,0 +1,405 @@
+	if err != nil {
+		t.Fatalf("Error listing nodes: %v", err)
+	}
+
+	for _, node := range nodes.Items {
+		nodeNames = append(nodeNames, node.Name)
+	}
+
+	daemonSet, err := clientset.AppsV1().DaemonSets("amazon-cloudwatch").Get(context.TODO(), "cloudwatch-agent", metav1.GetOptions{})
+	if err != nil {
+		t.Errorf("Error getting CloudWatch Agent DaemonSet: %v", err)
+	}
+	if daemonSet == nil {
+		t.Error("CloudWatch Agent DaemonSet not found")
+	}
+
+	configMap, err := clientset.CoreV1().ConfigMaps("amazon-cloudwatch").Get(context.TODO(), "cloudwatch-agent", metav1.GetOptions{})
+	if err != nil {
+		t.Errorf("Error getting CloudWatch Agent ConfigMap: %v", err)
+	}
+	if configMap == nil {
+		t.Error("CloudWatch Agent ConfigMap not found")
+	}
+
+	service, err := clientset.CoreV1().Services("amazon-cloudwatch").Get(context.TODO(), "cloudwatch-agent", metav1.GetOptions{})
+	if err != nil {
+		t.Errorf("Error getting CloudWatch Agent Service: %v", err)
+	}
+	if service == nil {
+		t.Error("CloudWatch Agent Service not found")
+	}
+
+	serviceAccount, err := clientset.CoreV1().ServiceAccounts("amazon-cloudwatch").Get(context.TODO(), "cloudwatch-agent", metav1.GetOptions{})
+	if err != nil {
+		t.Errorf("Error getting CloudWatch Agent Service Account: %v", err)
+	}
+	if serviceAccount == nil {
+		t.Error("CloudWatch Agent Service Account not found")
+	}
+}
+
+func TestMetrics(t *testing.T) {
+	env := environment.GetEnvironmentMetaData()
+	configFile := filepath.Base(env.AgentConfig)
+
+	tests, exists := testRegistry[configFile]
+	if !exists {
+		t.Skipf("No tests registered for config file: %s", configFile)
+		return
+	}
+
+	for _, testFunc := range tests {
+		testFunc(t)
+	}
+}
+
+func testTomcatMetrics(t *testing.T) {
+	t.Run("verify_jvm_tomcat_metrics", func(t *testing.T) {
+		metricsToCheck := []struct {
+			name      string
+			namespace string
+		}{
+			{"tomcat.traffic", "JVM_TOMCAT_E2E"},
+			{"jvm.classes.loaded", "JVM_TOMCAT_E2E"},
+			{"jvm.gc.collections.count", "JVM_TOMCAT_E2E"},
+			{"jvm.gc.collections.elapsed", "JVM_TOMCAT_E2E"},
+			{"jvm.memory.heap.init", "JVM_TOMCAT_E2E"},
+			{"jvm.memory.heap.max", "JVM_TOMCAT_E2E"},
+			{"jvm.memory.heap.used", "JVM_TOMCAT_E2E"},
+			{"jvm.memory.heap.committed", "JVM_TOMCAT_E2E"},
+			{"jvm.memory.nonheap.init", "JVM_TOMCAT_E2E"},
+			{"jvm.memory.nonheap.max", "JVM_TOMCAT_E2E"},
+			{"jvm.memory.nonheap.used", "JVM_TOMCAT_E2E"},
+			{"jvm.memory.nonheap.committed", "JVM_TOMCAT_E2E"},
+			{"jvm.memory.pool.init", "JVM_TOMCAT_E2E"},
+			{"jvm.memory.pool.max", "JVM_TOMCAT_E2E"},
+			{"jvm.memory.pool.used", "JVM_TOMCAT_E2E"},
+			{"jvm.memory.pool.committed", "JVM_TOMCAT_E2E"},
+			{"jvm.threads.count", "JVM_TOMCAT_E2E"},
+			{"tomcat.sessions", "JVM_TOMCAT_E2E"},
+			{"tomcat.errors", "JVM_TOMCAT_E2E"},
+			{"tomcat.request_count", "JVM_TOMCAT_E2E"},
+			{"tomcat.max_time", "JVM_TOMCAT_E2E"},
+			{"tomcat.processing_time", "JVM_TOMCAT_E2E"},
+			{"tomcat.threads", "JVM_TOMCAT_E2E"},
```

Comment:
can we do this in a more elegant way if the namespace is the same for all of these metrics? do we expect this to change in the future?

Reply:
Yeah good point, I can make it a constant.